server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -108,19 +122,41 @@ public Task(
      *            generate data?
      */
     public final TaskInfo taskInfo(String localNodeId, boolean detailed) {
+        return taskInfo(localNodeId, detailed, !detailed);
```

Comment:
This is here why? We always want to provide the stats when calling this method regardless of the caller asking for details?

```suggestion
        return taskInfo(localNodeId, detailed, detailed == false);
```

explicit false check is easier to read than boolean negation

Reply:
Modified to ```detailed == false```

Detailed flag is used to control including the resource stats information. The other method takes an input to specifically **exclude** resource information - used to exclude resource info during storing tasks in task index. 
```taskInfo(.. boolean detailed, boolean excludeResourceStats)```